src/internal/sio_client_impl.cpp

```diff
@@ -539,11 +547,17 @@ namespace sio
             this->m_client.send(this->m_con, *payload, frame::opcode::text);
         });
 
-        if(m_ping_timeout_timer)
+        if (m_ping_timeout_timer) {
+            asio::error_code ec;
+            m_ping_timeout_timer->expires_from_now(milliseconds(m_ping_interval + m_ping_timeout),ec);
+            m_ping_timeout_timer->async_wait(std::bind(&client_impl::timeout_pong, this, std::placeholders::_1));
+        }
```

Comment:
Why are you recreating the timer here instead of just cancelling it?

Reply:
When the on_ping event is received, the timer needs to be reset. If the on_ping event is not received before the next timeout, it is considered that the local network is disconnected.